hostname-debian: set main program and use in more places

mweinelt opened the pull request:
The debian hostname utility is a small replacement for the version from net-tools. The net-tools suite has been badly maintained since at least a decade, so we should phase out its use.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mweinelt committed:
hostname-debian: set main program

mweinelt committed:
nixos/ec2-data: use hostname-debian to set hostname from metadata

This is the simpler, more lightweight and better maintained version of
the hostname utility.

mweinelt committed:
nixos/zfs: use hostname-debian for hostname lookups

This is the simpler, more lightweight and better maintained version of
the hostname utility.

SuperSandro2000 commented:
We probably want to add https://github.com/NixOS/nixpkgs/blob/ee63ec601c0c6c5c09d399f087a4aa846461a563/pkgs/by-name/xd/xdg-utils/package.nix#L209

mweinelt committed:
xdg-utils: use hostname-debian for hostname lookups

This is the simpler, more lightweight and better maintained version of
the hostname utility.

mweinelt committed:
hostname-debian: 3.23 -> 3.25

https://salsa.debian.org/meskes/hostname/-/blob/debian/3.25/debian/changelog

adamcstephens commented:
Is this failure from the tests relevant?

```
error: 1 dependencies of derivation '/nix/store/zix6dvl49a8bn6blxjicknq47kikk1iv-xdg-mime-test.drv' failed to build
```

@ofborg test zfs_2_3

mweinelt committed:
xdg-utils: drop kernelkisser.png mirror that redirects to archive.org

This leads to a different hash, thereby breaking the test.

mweinelt commented:
Yes, please limit it to whatever platforms wlroots supports.

adamcstephens commented:
Or alternatively: https://cgit.freebsd.org/src/tree/bin/hostname

MattSturgeon commented:
> > Obviously, it was probably not expected for `wl-copy` to build at all on darwin.
>
> Yes, please limit it to whatever platforms wlroots supports.

It seems like darwin support was intentionally added in [`11fc036` (#224878)](https://github.com/NixOS/nixpkgs/pull/224878/commits/11fc036357c11d5a09d1a8e93c6a62c604febab1).

Maybe @wegank has some insight for when/why wl-clipboard would be needed on darwin?
